google/cloud/internal/opentelemetry.h

```diff
@@ -30,12 +32,74 @@ namespace internal {
 
 bool TracingEnabled(Options const& options);
 
+/**
+ * Returns a [tracer] to use for creating [spans].
+ *
+ * This function exists for the sake of testing. Library maintainers should call
+ * `MakeSpan(...)` directly to create a span.
+ *
+ * @see https://opentelemetry.io/docs/instrumentation/cpp/manual/#initializing-tracing
+ *
+ * [spans]:
+ * https://opentelemetry.io/docs/concepts/signals/traces/#spans-in-opentelemetry
+ * [tracer]: https://opentelemetry.io/docs/concepts/signals/traces/#tracer
+ */
 opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> GetTracer(
     Options const& options);
 
+/**
+ * Start a [span] using the current [tracer].
+ *
+ * The current tracer is determined by the prevailing `CurrentOptions()`.
+ *
+ * @see https://opentelemetry.io/docs/instrumentation/cpp/manual/#start-a-span
+ *
+ * [span]:
+ * https://opentelemetry.io/docs/concepts/signals/traces/#spans-in-opentelemetry
+ * [tracer]: https://opentelemetry.io/docs/concepts/signals/traces/#tracer
+ */
 opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
     opentelemetry::nostd::string_view name);
 
+/**
+ * Extracts information from a cloud status and adds it to a span.
+ *
+ * This method will end the span, and set its [span status], accordingly. Other
```

Comment:
Ending the span seems like it might be important enough to have in the function name.

Reply:
Ok, we have a `MakeSpan()`, so I will just call the thing `EndSpan()`.